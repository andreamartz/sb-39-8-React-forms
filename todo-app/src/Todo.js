/**
 * Todo:
 *   * display a divv with the task of the todo
 */

import React from "react";

const Todo = ({ id, task, removeTodo }) => {
  const handleRemove = () => {
    removeTodo(id);
  }
  return (
    <li id={`${id}`}>{task} <span onClick={handleRemove}> X </span></li>
  )
}

export default Todo;
