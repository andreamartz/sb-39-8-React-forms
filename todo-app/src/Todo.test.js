import React from "react";
import { render } from '@testing-library/react';
import Todo from './Todo';

describe("Smoke tests", () => {
  test('renders without crashing', () => {
    render(<Todo />);
  });
});

describe("Snapshot tests", () => {
  test('matches snapshot', () => {
    const {asFragment} = render(<Todo />);
    expect(asFragment()).toMatchSnapshot();
  });
});