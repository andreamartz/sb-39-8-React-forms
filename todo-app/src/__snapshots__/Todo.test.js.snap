// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests matches snapshot 1`] = `
<DocumentFragment>
  <li
    id="undefined"
  >
    <span>
       X 
    </span>
  </li>
</DocumentFragment>
`;
